### ISSUE TITLE
Use of standardScaler() and TPOT

### ISSUE THREAD

**Issue by soliverc** [EpistasisLab/tpot#973]
It is good practice to first split the data into train and test, then scale by `fit_transform` on the training set, and `transform` on the test set. 

So I create a train test split:
```python
training_features, testing_features, training_target, testing_target = \
            train_test_split(data, labels, random_state=1,stratify=labels.values,test_size = 0.2)
```
Scale the data
```python
sc = StandardScaler()
training_features_scaled = sc.fit_transform(training_features)
testing_features_scaled = sc.transform(testing_features)
```

Now I feed `training_features_scaled` into `TPOT`.

However, TPOT then runs cross fold validation on this dataset, which bascially means that in this case, scaling has been done before the split and there's a potential for leakage between train and test during each fold. Is this something that we should avoid, and if so, how can we avoid this?

**Comment by weixuanfu** [comment]
How about a template like `StandardScaler-Selector-Transformer-Classifier`? Since `StandardScaler` is [a operator in TPOT default configuration](https://github.com/EpistasisLab/tpot/blob/v0.11.0/tpot/config/classifier.py#L169).

**Comment by soliverc** [comment]
Ok, The template I use is simply `Classifier`. I could add `Standardscaler` to the list. 

But how do I then scale the `testing_features` set that TPOT hasn't seen whatsoever?

**Comment by weixuanfu** [comment]
To avoild potential leakage, I don't think you need scale the `testing_features` any more if the template is 'Standardscaler-Classifier'. Because the output pipeline from TPOT has `StandardScaler` as first step and can be directly used on the non-scaled `testing_features`. Is this what you need?

**Comment by soliverc** [comment]
Ok, I'll test that out. Doesn't the standard scaler used on the unseen `testing_features` have to be first fitted to the `training_features`? So how does this scaler that was fit to the training features survive the process so I can use it again? Or can I just simply `fit_transform` on the unseen data?

**Comment by weixuanfu** [comment]
Yes, just deleting the codes below and fit `training_features` directly in TPOTClassifier with `template='StandardScaler-Classifier'`. 
```python
sc = StandardScaler()
training_features_scaled = sc.fit_transform(training_features)
testing_features_scaled = sc.transform(testing_features)
```

Then the scaler should fitted on the training set in each cv fold for avoid leakage during TPOT optimization . Then the optimized pipeline should be fitted on `training_features` and can be used for the unseen `testing_features.`. If you prefer to check the transformed unseen  `testing_features`, please try` tpot_obj.fitted_pipeline_.steps[0][1].fit_transform(testing_features)`.
